_blogposts/2024-02-01-release-11-1-0.mdx

```diff
@@ -0,0 +1,149 @@
+
+```rescript
+// Greet.res
+
+// @jsx.component works the same as @react.component does in React
+@jsx.component
+let make = (~name) => {
+  <div>
+    {Preact.string("Hello " ++ name)}
+  </div>
+}
+```
+
+Read more in the [`new documentation on the generic JSX transform`](/docs/manual/latest/jsx#generic-jsx-transform-jsx-beyond-react-experimental).
+
+## Tagged template literals
+
+This release comes with support for [tagged templates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates).
+
+A tag function in JavaScript is a function that expects an array of strings and variadic parameters as input. Now it's possibe to bind to such functions with the new [`@taggedTemplate`](/syntax-lookup#taggedTemplate-decorator) decorator:
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```rescript
+// see https://bun.sh/docs/runtime/shell
+type result = {exitCode: int}
+@module("bun") @taggedTemplate
+external sh: (array<string>, array<string>) => promise<result> = "$"
+
+let filename = "index.res"
+let result = await sh`ls ${filename}`
+```
+```js
+var $$Bun = require("bun");
+
+var filename = "index.res";
+
+var result = await $$Bun.$`ls ${filename}`;
+```
+
+</CodeTab>
+
+
+Of course you can also create your own tag functions in ReScript now as well, it is just a function with the following signature.
+
+```rescript
+let myTagFunction : (array<string>, array<'param>) => 'output
+```
+
+Refer to the docs to find a [detailed example](/docs/manual/latest/tagged-templates).
+
+## Import attributes
+
+Import attributes is a JS feature that is [currently in standardization](https://github.com/tc39/proposal-import-attributes), but is already implemented by many JS tools. Now, ReScript supports it too, as long as the compiler is configured to output ES6.
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+@module({from: "./myJson.json", with: {type_: "json", \"some-identifier": "yep"}})
+external myJson: Js.Json.t = "default"
+Console.log(myJson)
+
+@module({from: "./myCss.css", with: {type_: "css", \"some-identifier": "yep"}})
+external buttonCss: string = "button"
+Console.log(buttonCss)
+```
+```js
+import * as MyCssCss from "./myCss.css" with {"type": "css", "some-identifier": "yep"};
+import MyJsonJson from "./myJson.json" with {"type": "json", "some-identifier": "yep"};
+
+var myJson = MyJsonJson;
+console.log(myJson);
+
+var buttonCss = MyCssCss.button;
+console.log(buttonCss);
+```
+</CodeTab>
+
+## Array spread syntax
+
+The spread syntax, which was already supported for records and lists for a long time, now also supports arrays!
+
+```rescript
+let animals = ["🐶", "🐱", "🐷"]
+let moreAnimals = [...animals, "🐔", "🐴", "🐮"]
+````
+
+## Hyphens in JSX tag names
+
+We lifted restrictions on JSX tag names. This means you no longer need to escape tag names that contain hyphens:
+
+Previously:
+
+```rescript
+let x = <\"custom-tag" />
+let y = <Foo.\"custom-tag" />
```

Comment:
to be honest, I'm not sure of the semantics behind a lowercase tag from a module, we might as well remove this example don't you think?

Reply:
Agreed.